array_api_tests/test_elementwise_functions.py

```diff
@@ -374,9 +374,19 @@ def test_divide(args):
     # could test that this does implement IEEE 754 division, but we don't yet
     # have those sorts in general for this module.
 
-@given(two_any_dtypes.flatmap(lambda i: two_array_scalars(*i)))
-def test_equal(args):
-    x1, x2 = args
+
+
+@given(
+    x1=xps.arrays(
+        dtype=shared_dtypes,
+        shape=shared(two_broadcastable_shapes(), key="shape_pair").map(lambda pair: pair[0])
+    ),
+    x2=xps.arrays(
+        dtype=promotable_dtypes(shared_dtypes),
+        shape=shared(two_broadcastable_shapes(), key="shape_pair").map(lambda pair: pair[1])
```

Comment:
This is complicated enough that we should factor it out into a helper function. Maybe something like

```py
@composite
def two_arrays(draw, dtypes=dtypes):
    # return two arrays with promotable dtypes drawn from dtypes and mutually broadcastable shapes
```

and we can then use 

```py
@given(*two_arrays(dtypes)) # replacing dtypes with the appropriate strategy
```

Also note that this should be two_mutually_broadcastable_shapes, not two_broadcastable_shapes. The latter only creates shapes that broadcast to the first shape. We should probably rename it to something clearer.

Reply:
Good idea thanks. I think I'll revert/drop all the `test_elementwise_functions.py` changes in this PR, as it'll get annoying working on my other one.